### PR TITLE
REF-73: Delivery execution state controller

### PR DESCRIPTION
**Pull request by wazelin**
- [REF-73](https://oat-sa.atlassian.net/browse/REF-73) fix(readme): add a text to the Open API documentation link
- [REF-73](https://oat-sa.atlassian.net/browse/REF-73) fix(api-specification): add authentication specification
- [REF-73](https://oat-sa.atlassian.net/browse/REF-73) chore(api-specification): extract responses
- [REF-73](https://oat-sa.atlassian.net/browse/REF-73) chore(api-specification): add `PUT /DeliveryState` specification
- [REF-73](https://oat-sa.atlassian.net/browse/REF-73) feat(delivery-execution): implement PUT `DeliveryState` request handler
 - [REF-73](https://oat-sa.atlassian.net/browse/REF-73) chore(version): bump a minor one

[The endpoint's documentation](https://openapi.taotesting.com/viewer#/?url=https://github.com/oat-sa/extension-tao-delivery/blob/feature/REF-73/delivery-execution-state-controller/doc/rest.json).

[Backport](https://github.com/oat-sa/extension-tao-delivery/pull/456).

### CODE REVIEW

**Comment by Yury-Razhkou** [comment]
Old `unstop` method changing status from `TERMINATED` to `ACTIVE` and trigger `DeliveryExecutionReactivated` event. New method allows doing the same without triggering this event. Almost the same thing with changing from `FINISHED` to `ACTIVE` (https://github.com/oat-sa/extension-tao-proctoring/blob/a448f14f7dd7d76296157c8b715f0f04b1273592/model/execution/DeliveryExecutionManagerService.php#L185). Not sure how this will work in our case. But i think we need input from somebody who knows this part better.

**Comment by wazelin** [comment]
@Yury-Razhkou, I don't really think we need that, at least for our case.
The events are needed there for proctored delivery executions.

**Comment by wazelin** [comment]
OK, double checked that, and you are absolutely right @Yury-Razhkou. My apologies.

That event handler actually sets proper data to the test session. Tricky.

**Comment by wazelin** [comment]
All right, triple checked now :D
Turns out all the events are being sent correctly, except for a very specific `unstop` action's `DeliveryExecutionReactivated`, but IMO we don't even need it in our use case.

If a delivery manager / proctor wants to unstop the execution, they should call a specific `unstop` method. It will do the same as this endpoint with `DeliveryExecutionStatusPaused`, and additionally will set a `suspended` test session state.

File: controller/DeliveryExecutionState.php [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    {
+        /** @noinspection PhpIncompatibleReturnTypeInspection */
+        return $this->getServiceLocator()->get(StateServiceInterface::SERVICE_ID);
+    }
+
+    private function extractState(): string
+    {
+        $data = json_decode((string)$this->getPsrRequest()->getBody(), true);
+
+        if (empty($data['value']) || !in_array($data['value'], $this->getStateService()->getDeliveriesStates(), true)) {
```
Comment (ktsirkou) [inline controller/DeliveryExecutionState.php:89]: Due to 

```
    public function getDeliveriesStates()
    {
         return [
             DeliveryExecution::STATE_FINISHED,
             DeliveryExecution::STATE_ACTIVE,
             DeliveryExecution::STATE_PAUSED
         ];
    }
```

We will not be able to set state Terminated, is there a reason for avoid this?

Comment (wazelin) [reply controller/DeliveryExecutionState.php:89]: That's the restriction of the current service.
We can extend it if needed later IMO.

So far I consider setting the delivery execution to a `TERMINATED` state to be covered by YAGNI :)

**Pull request merged**